modules/SRanipal_module/src/SRanipalModule.cpp

```diff
@@ -0,0 +1,360 @@
+                return false;
+            }
+
+        } else
+        {
+            yarp::os::Property rcb_face_conf{
+                {"device", yarp::os::Value("remote_controlboard")},
+                {"local", yarp::os::Value("/" + name + "/face/remoteControlBoard")},
+                {"remote", yarp::os::Value("/" + robot + "/face")},
+                {"part", yarp::os::Value("face")}};
+
+            if (m_poly.open(rcb_face_conf))
+            {
+                yarp::dev::IControlMode* iCM{nullptr};
+                yarp::dev::IControlLimits* iCtrlLim{nullptr};
+                bool ok = m_poly.view(iCM);
+                ok &= m_poly.view(m_iPos);
+                ok &= m_poly.view(iCtrlLim);
+                if (iCM)
+                    ok &= iCM->setControlMode(0, VOCAB_CM_POSITION);
+                if (iCtrlLim)
+                {
+                    ok &= iCtrlLim->getLimits(0, &m_minEyeLid, &m_maxEyeLid);
+                    m_maxEyeLid = 0.9 * m_maxEyeLid;
+                }
+                if (m_iPos)
+                {
+                    ok &= m_iPos->setRefSpeed(0, 75.0); // max velocity that doesn't give problems
+                    ok &= m_iPos->setRefAcceleration(0, std::numeric_limits<double>::max());
+                }
+                if (!ok)
+                {
+                    yError() << "Failed to configure the remote_controlboard";
+                    return false;
+                }
+            } else
+            {
+                yError() << "Failed to connect to the face control board. Set useRawEyelids to "
+                            "true to avoid connecting to it.";
+                return false;
+            }
+        }
+
+        yInfo() << "Using Eye tracking.";
+    }
+
+    if (m_useLip)
+    {
+        output = ViveSR::anipal::Initial(ViveSR::anipal::Lip::ANIPAL_TYPE_LIP, NULL); //LIP V1 works better in detecting the smile
+        if (output != ViveSR::Error::WORK) {
+            yError("[SRanipalModule::configure] Failed to initialize Lip engine [%d - %s].", output, errorCodeToString(output));
+            return false;
+        }
+
+        std::string lipImageOutputPort = rf.check("lipImagePortName", yarp::os::Value("/lipImage:o"), "The name of the output port for the lip camera.").asString();
+        if (!m_lipImagePort.open("/" + name + lipImageOutputPort))
+        {
+            yError() << "[SRanipalModule::configure] Failed to open /" + name + lipImageOutputPort + " port.";
+            return false;
+        }
+
+        yInfo() << "Using Lip tracking.";
+    }
+
+    yInfo() << "SRanipalModule started correctly.";
+
+    return true;
+}
+
+double SRanipalModule::getPeriod()
+{
+    return m_period;
+}
+
+bool SRanipalModule::updateModule()
+{
+    std::lock_guard<std::mutex> lock(m_mutex);
+
+    ViveSR::anipal::Eye::EyeData_v2 eye_data_v2;
+    ViveSR::anipal::Lip::LipData lip_data; //The lip data V1 works better in detecting the smile
+    lip_data.image = m_lipImage;
```

Comment:
Can you explain this line of code? probably because ` lip_data.image` was not initialized, and we needed to initialize it in order to receive images from the lip_data structure (lip camera)?

Reply:
``lip_data.image`` is a pointer. We need to specify the pointer to the container where we will store the image. So yes, in a sense we need to initialize it.